sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -176,10 +173,16 @@
     Objects.requireNonNull(key, "key");
     if (value == null
         || (value.getType() == AttributeValue.Type.STRING && value.getStringValue() == null)) {
+      if (attributes == null) {
+        return this;
+      }
       attributes.remove(key);
-    } else {
-      attributes.put(key, value);
+      return this;
```

Comment:
Could be simplified to `if null { remove } return;`

Reply:
Done.